examples/autoqasm/3_2_magic_state_distillation.ipynb

```diff
@@ -0,0 +1,447 @@
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "3ba7d657-f5d9-4933-a91b-a870d565d9d4",
+   "metadata": {},
+   "source": [
+    "Let's first import the modules used in this notebook."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "id": "ccdb0127-3c13-42b1-980c-c70f2e16fc00",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# general imports\n",
+    "from typing import Dict, List\n",
+    "from collections import Counter\n",
+    "import numpy as np\n",
+    "from collections import defaultdict\n",
+    "\n",
+    "# AWS imports: Import Braket SDK modules\n",
+    "from braket.devices.local_simulator import LocalSimulator\n",
+    "import braket.experimental.autoqasm as aq\n",
+    "import braket.experimental.autoqasm.instructions as ins"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "6f902b86-4201-4d0c-a371-6482743f67be",
+   "metadata": {},
+   "source": [
+    "## Gate teleportation\n",
+    "To form a universal gateset, at least one non-Clifford gate is needed in addition to Clifford gates. This additional gate is commonly chosen to be $RZ(\\pi/4)$ or $RZ(\\pi/6)$ gate. These non-Clifford gates are not possible to implement transversally with stabilizer code. Including these non-Clifford gates would lose the desirable properties of Clifford gates. Here is where magic states come to the rescue. In this section, let's learn how to apply a $RZ(\\pi/6)$ gate with only Clifford gates and a magic state [1], a procedure known as gate teleportation. For simplicity, we first focus on demonstrating the algorithm of gate teleportation with a physical circuit, i.e., without encoding to an error correction code. The same algorithm can also apply to logical circuits, by replacing the Clifford gates and the magic state with the logical version. After this section, we will introduce how to create a logical magic state."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "d2f60849-0d78-4dff-93d8-7213e2ead652",
+   "metadata": {},
+   "source": [
+    "First, we define a subroutine, `physical_magic_state`, to create a physical magic state, $\\ket{A_{\\pi/6}} = \\frac{1}{\\sqrt{2}}(\\ket{0}+e^{i\\pi/6}\\ket{1})$. Because the example will be executed on an ideal simulator, the subroutine creates an ideal magic state. But on a near-term hardware, this magic state preparation would be subjected to noise.\n",
```

Comment:
```suggestion
    "First, we define a subroutine, `physical_magic_state`, to create a physical magic state, $\\ket{A_{\\pi/6}} = \\frac{1}{\\sqrt{2}}(\\ket{0}+e^{i\\pi/6}\\ket{1})$. Because the example will be executed on an ideal simulator, the subroutine creates an ideal magic state. But on a near-term hardware, this magic state preparation would be subject to noise.\n",
```